media-source-respec.html

```diff
@@ -404,108 +411,202 @@ <h3>Attributes</h3>
-                <li>For each {{AudioTrack}} object in the <var>SourceBuffer audioTracks list</var>, run the following steps:
+                <li>For each {{AudioTrack}} object in the |SourceBuffer audioTracks list|, run the following steps:
                   <ol>
                     <li>Set the {{AudioTrack/sourceBuffer}} attribute on the {{AudioTrack}} object to null.</li>
-                    <li>Remove the {{AudioTrack}} object from the <var>HTMLMediaElement audioTracks list</var>.
+
+                    <li>Remove the {{AudioTrack}} object from the |SourceBuffer audioTracks list|.
                     <p class="note">
-                      This should trigger {{AudioTrackList}} [[HTML]] logic to <a def-id="queue-and-fire-media-removetrack-with-track-attr-initialized-to"></a> the {{AudioTrack}} object, at the <var>HTMLMediaElement audioTracks list</var>.
-                      If the {{AudioTrack/enabled}} attribute on the {{AudioTrack}} object was true at the beginning of this removal step, then this should also trigger {{AudioTrackList}} [[HTML]] logic to [=queue a task=] to [=fire an event=] named <a def-id="mediatracklist-change"></a> at the <var>HTMLMediaElement audioTracks list</var>
+                      This should trigger {{AudioTrackList}} [[HTML]] logic to <a
+                        def-id="queue-and-fire-media-removetrack-with-track-attr-initialized-to"></a> the {{AudioTrack}}
+                      object, at the |SourceBuffer audioTracks list|.  If the {{AudioTrack/enabled}} attribute on the
+                      {{AudioTrack}} object was true at the beginning of this removal step, then this should also
+                      trigger {{AudioTrackList}} [[HTML]] logic to [=queue a task=] to [=fire an event=] named <a
+                        def-id="mediatracklist-change"></a> at the |SourceBuffer audioTracks list|.
                     </p>
                     </li>
-                    <li>Remove the {{AudioTrack}} object from the <var>SourceBuffer audioTracks list</var>.
-                    <p class="note">
-                      This should trigger {{AudioTrackList}} [[HTML]] logic to <a def-id="queue-and-fire-media-removetrack-with-track-attr-initialized-to"></a> the {{AudioTrack}} object, at the <var>SourceBuffer audioTracks list</var>.
-                      If the {{AudioTrack/enabled}} attribute on the {{AudioTrack}} object was true at the beginning of this removal step, then this should also trigger {{AudioTrackList}} [[HTML]] logic to [=queue a task=] to [=fire an event=] named <a def-id="mediatracklist-change"></a> at the <var>SourceBuffer audioTracks list</var>
-                    </p>
+
+                    <li>
+                      <dl class="switch">
+                        <dt>If the {{MediaSource}} was constructed in a {{DedicatedWorkerGlobalScope}}:</dt>
```

Comment:
> IIUC, the purpose of the mirror messages is to ensure that the HTML Media Element update steps run as their own task on the main event loop, rather than the associated changes happening "magically" in the middle of some other task execution, right ?

This is typically the case, yes. Especially will be more so once I update this to likely limit any optimizations of the 'cross-context communication mechanism' for web platform security and interop.

Reply:
> This is by no means an essential change, but I think this would be clearer if the "Otherwise" steps below were broken out an given a name (e.g. "_Remove an audio track from Media Element audio track list_") and then this step can be something like
> 
> > If the MediaSource was constructed in a DedicatedWorkerGlobalScope, post an internal remove track mirror message ... whose implicit handler runs the _Remove an audio track from Media Element audio track list_ algorithm on ....
> > Otherwise run the _Remove an audio track from Media Element audio track list_ algorithm on ....

I'm looking to see if there's a way to keep such a "named step" within the method algorithm steps instead of being moved to an explicit algorithm in the SourceBuffer algorithm's section. Do you have any preference on this?

Reply:
For consistency with other media WG specs, and internal consistency with existing MSE spec, moving such a "named step" to be an explicit algorithm is probably best.

Reply:
Mirror if necessary seems the better approach for this, as it is more general.